File: Modul-4/Soal-4/PRAK404-2410817210027-HAFIZPERDANA.py
def penjumlahan(a, b):
    return a + b

def pengurangan(a, b):
    return a - b

def perkalian(a, b):
    return a * b

def pembagian(a, b):
    return a / b

while True:
    print("Pilih program:")
    print("1. Penjumlahan")
    print("2. Pengurangan")
    print("3. Perkalian")
    print("4. Pembagian")
    print("5. Exit")
    
    pilihan = int(input("Masukkan pilihan: "))

    if pilihan == 5:
        print("Terimakasih, telah menggunakan kalkulator Hafiz Perdana")
        break

    if pilihan < 1 or pilihan > 5:
        print("Input anda salah, silakan coba lagi")
        continue
    
    angka1 = float(input("Masukkan nilai pertama: "))
    angka2 = float(input("Masukkan nilai kedua: "))
    
    if pilihan == 1:
        hasil = penjumlahan(angka1, angka2)
        print(f"Hasil penjumlahan antara {angka1:.2f} dengan {angka2:.2f} adalah {hasil:.2f}")
    elif pilihan == 2:
        hasil = pengurangan(angka1, angka2)
        print(f"Hasil pengurangan antara {angka1:.2f} dengan {angka2:.2f} adalah {hasil:.2f}")
    elif pilihan == 3:
        hasil = perkalian(angka1, angka2)
        print(f"Hasil perkalian antara {angka1:.2f} dengan {angka2:.2f} adalah {hasil:.2f}")
    elif pilihan == 4:
        hasil = pembagian(angka1, angka2)
        print(f"Hasil pembagian antara {angka1:.2f} dengan {angka2:.2f} adalah {hasil:.2f}")

File: Modul-4/Soal-4/PRAK404-2410817210027-HAFIZPERDANA.c
#include <stdio.h>

int main() {
    int pilihan;
    float angka1, angka2, hasil;

    do {
        printf("Pilih program:\n");
        printf("1. Penjumlahan\n");
        printf("2. Pengurangan\n");
        printf("3. Perkalian\n");
        printf("4. Pembagian\n");
        printf("5. Exit\n");
        printf("Masukkan pilihan: ");
        scanf("%d", &pilihan);

        switch(pilihan) {
            case 1:
                printf("Masukkan nilai pertama: ");
                scanf("%f", &angka1);
                printf("Masukkan nilai kedua: ");
                scanf("%f", &angka2);
                hasil = angka1 + angka2;
                printf("Hasil penjumlahan antara %.2f dengan %.2f adalah %.2f\n", angka1, angka2, hasil);
                break;
            case 2:
                printf("Masukkan nilai pertama: ");
                scanf("%f", &angka1);
                printf("Masukkan nilai kedua: ");
                scanf("%f", &angka2);
                hasil = angka1 - angka2;
                printf("Hasil pengurangan antara %.2f dengan %.2f adalah %.2f\n", angka1, angka2, hasil);
                break;
            case 3:
                printf("Masukkan nilai pertama: ");
                scanf("%f", &angka1);
                printf("Masukkan nilai kedua: ");
                scanf("%f", &angka2);
                hasil = angka1 * angka2;
                printf("Hasil perkalian antara %.2f dengan %.2f adalah %.2f\n", angka1, angka2, hasil);
                break;
            case 4:
                printf("Masukkan nilai pertama: ");
                scanf("%f", &angka1);
                printf("Masukkan nilai kedua: ");
                scanf("%f", &angka2);
                hasil = angka1 / angka2;
                printf("Hasil pembagian antara %.2f dengan %.2f adalah %.2f\n", angka1, angka2, hasil);
                break;
            case 5:
                printf("Terimakasih, telah menggunakan kalkulator Hafiz Perdana\n");
                break;
            default:
                printf("Input anda salah, silakan coba lagi\n");
        }
    } while (pilihan != 5);

    return 0;
}